### DISCUSSION TITLE
`None` for Scalar type instead of error in the response

### DISCUSSION THREAD

**Discussion by 1g0rkR** [async-graphql/async-graphql#1205] [Q&A]
Hi there,

How could I setup a Scalar type to have `None` as a result for errors? There is smth when I use Option for the argument and passing `null` as a value. I want to have None manually.

```
#[derive(Clone, PartialEq, Debug)]
struct Param(String);

#[Scalar]
impl ScalarType for Param {
    fn parse(value: Value) -> InputValueResult<Self> {
        match value {
            Value::String(s) if !s.is_empty() => Ok(Self(s)),
            _ => *** None here ***,
        }
    }

    fn to_value(&self) -> Value {
        Value::String(self.0.clone())
    }
}
```

And QueryRoot:
```
#[derive(Default)]
struct QueryRoot;

#[Object]
impl QueryRoot {
    async fn test(&self, _ctx: &Context<'_>, param: Option<Param>) -> Result<bool> {
        println!("{:?}", param);
        Ok(true)
    }
}
```

**Comment by sunli829** [comment]
You just need to return an error when the `Param` cannot be parsed.

```rust
#[Scalar]
impl ScalarType for Param {
    fn parse(value: Value) -> InputValueResult<Self> {
        match value {
            Value::String(s) if !s.is_empty() => Ok(Self(s)),
            _ => Err(InputValueError::expected_type(value)), // returns an error
        }
    }
}
```

**Reply by 1g0rkR** [reply]
It makes sense for non-optional argument. What about Option argument? I am trying to have None for all cases except string is not null but scalar returns an error instead of go ahead with None for argument.

<img width="1423" alt="image" src="https://user-images.githubusercontent.com/90048102/212490997-b4cd466d-7502-44d6-b08b-3dd954aaa8d7.png">
